MH4U Dex/CoreDataController.m

```diff
@@ -0,0 +1,430 @@
+            return nil;
+        }
+    } else {
+        NSLog(@"Error fetching object!");
+        NSLog(@"%@, %@", error, error.localizedDescription);
+        return nil;
+    }
+
+}
+
+- (Area *)getUniqueAreaWithCombinedName:(NSString *)combinedName
+{
+    NSError *error = nil;
+    NSFetchRequest *request = [[NSFetchRequest alloc] initWithEntityName:@"Area"];
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K == %@", @"combinedName", combinedName];
+    [request setPredicate:predicate];
+    
+    NSArray *results = [self.managedObjectContext executeFetchRequest:request error:&error];
+    if (!error) {
+        if (results.count == 0) {
+            // Object not found.
+            return nil;
+        } else if (results.count == 1) {
+            return results.firstObject;
+        } else {
+            NSLog(@"Too many entities found!");
+            return nil;
+        }
+    } else {
+        NSLog(@"Error fetching object!");
+        NSLog(@"%@, %@", error, error.localizedDescription);
+        return nil;
+    }
+}
+
+- (NSManagedObject *)getUniqueEntityWithEntityName:(NSString *)entityName withNameAttribute:(NSString *)name
+{
+    NSError *error = nil;
+    NSFetchRequest *request = [[NSFetchRequest alloc] initWithEntityName:entityName];
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K == %@", @"name", name];
+    [request setPredicate:predicate];
+    
+    NSArray *results = [self.managedObjectContext executeFetchRequest:request error:&error];
+    if (!error) {
+        if (results.count == 0) {
+            // Object not found.
+            return nil;
+        } else if (results.count == 1) {
+            return results.firstObject;
+        } else {
+            NSLog(@"Too many entities found!");
+            return nil;
+        }
+    } else {
+        NSLog(@"Error fetching object!");
+        NSLog(@"%@, %@", error, error.localizedDescription);
+        return nil;
+    }
+}
+
+- (NSManagedObject *)getUniqueEntityWithEntityName:(NSString *)entityName withIDNumber:(NSNumber *)entity_id
+{
+    NSError *error = nil;
+    NSFetchRequest *request = [[NSFetchRequest alloc] initWithEntityName:entityName];
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K == %d", @"id", entity_id.intValue];
+    [request setPredicate:predicate];
+    
+    NSArray *results = [self.managedObjectContext executeFetchRequest:request error:&error];
+    if (!error) {
+        if (results.count == 0) {
+            return nil;
+        } else if (results.count == 1) {
+            return results.firstObject;
+        } else {
+            NSLog(@"Too many entities found!");
+            return nil;
+        }
+    } else {
+        NSLog(@"Error fetching object!");
+        NSLog(@"%@, %@", error, error.localizedDescription);
+        return nil;
+    }
+}
+
+- (NSUInteger)checkCountForEntityWithEntityName:(NSString *)entityName withNameAttribute:(NSString *)name
```

Comment:
Since this just returns the count and doesn't apply any testing/checking, I might name this `countForEntity...`.